solutions/gold/cf-1582F1.mdx

```diff
@@ -0,0 +1,104 @@
+---
+id: cf-1582F1
+source: CF
+title: Korney Korneevich and XOR (easy version)
+author: Chuyang Wang
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/96267)
+
+## Explanation
+
+To solve this problem, let's notice that the result of performing XOR
+operations on a sequence of $a_i < 500$ is strictly less than $2^9$. That's
+because the ninth bit is the largest possible in the binary representation of
+$a_i$. 
+
+Let $dp[x][i]$ represent the feasibility of getting $x$ as a result of XOR of
+an increasing sequence ending with a number less than or equal to $i$. There
+exists the following recurrence: 
+
+$$
+dp[x \oplus a_i][i] = dp[x \oplus a_i][0..i] \lor dp[x][0..i-1]
+$$
+
+with 
+
+$$
+dp[0][i] = \text{true}
+$$
+
+since it is always possible to get 0 from an increasing sequence consisting of
+zero element. 
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(n \cdot 2^{\lceil \log_2(\text{MAXA}) \rceil})$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+const int MAXA = 512;
+
+int main() {
+	int n;
+	cin >> n;
+	vector<int> arr(n);
+	for (int &i : arr) {
+		cin >> i;
+	}
+
+	/*
+	 * dp[x][i] := whether it is possible to get x using XOR of an increasing
```

Comment:
given that every other dp in code comments uses normal equal sign, i'd do away w/ the colon here (optional)